### PR TITLE
Zoom using FoV

### LINKED ISSUE #542: Camera can enter object
### Description
Really there are two problems here. The first is the near and far plane are set arbitrarily to 0.1 and 10 times the framed height, respectively, which is not enough in all cases. More importantly, zooming the camera moves it physically closer to the center of model, and thus sometimes inside of the model, which can't be fixed by the near plane. I would propose solving this by zooming in by narrowing the FOV and keeping the camera at the ideal radius. 

#### Live Demo
https://maps3d.io/blog/
Just zoom in and you'll soon see something like this:
Screenshot from 2019-05-09 09-03-50

### Browser Affected

- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Helios
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [ ] iOS
- [x] Linux
- [ ] MacOS
- [ ] Windows

### PR DESCRIPTION
**Pull request by elalish**
Fixes #542 

In addition to zooming using FoV instead of radius, I also interpolate using log(FoV) in order to make it perpetually linear (same movement is the same fractional change from any starting point). I also narrowed the minimum so you can get closer to the object, since you can never enter it now.

### CODE REVIEW

File: src/test/features/controls-spec.ts [hunk 1]
```diff
@@ -149,7 +149,7 @@ suite('ModelViewerElementBase with ControlsMixin', () => {
       });
 
       test('defaults FOV correctly', async () => {
-        expect(element.getFieldOfView()).to.be.equal(DEFAULT_FOV);
+        expect(element.getFieldOfView()).to.be.closeTo(DEFAULT_FOV, 0.00001);
```
Comment (cdata) [inline src/test/features/controls-spec.ts]: 👯‍♂

File: src/three-components/SmoothControls.ts [hunk 2]
```diff
@@ -35,9 +35,9 @@ export interface SmoothControlsOptions {
   // The maximum angle between model-forward and the camera azimuthal position
   maximumAzimuthalAngle?: number;
   // The minimum camera field of view in degrees
-  minimumFov?: number;
+  minimumLogFov?: number;
```
Comment (cdata) [inline src/three-components/SmoothControls.ts]: `fieldOfView`

Comment (cdata) [reply src/three-components/SmoothControls.ts]: Separately, its incongruous for us to ask the user to think about "log field of view" here when configuring the controls, but separately set field of view directly in a method like `setFov`. It seems like we could get away with asking the user to set a desired range in degrees and we could `log(fov)` as an internal detail.

File: src/test/three-components/SmoothControls-spec.ts [hunk 3]
```diff
@@ -280,13 +280,17 @@ suite('SmoothControls', () => {
           test('does not zoom when scrolling while blurred', () => {
```
Comment (cdata) [inline src/test/three-components/SmoothControls-spec.ts]: Let's add test coverage for configuring min/max field of view.

File: src/three-components/SmoothControls.ts (hunk 2 above)
Comment (elalish) [reply src/three-components/SmoothControls.ts]: Good call.

File: src/test/three-components/SmoothControls-spec.ts [hunk 4]
```diff
@@ -237,6 +237,25 @@ suite('SmoothControls', () => {
         });
       });
 
+      suite('field of view', () => {
```
Comment (cdata) [inline src/test/three-components/SmoothControls-spec.ts]: 💯

**Comment by pushmatrix** [comment]
Amazing!!

**Pull request merged**